pkg/controller/androidvariant/androidvariant_controller.go

```diff
@@ -124,32 +125,48 @@ func (r *ReconcileAndroidVariant) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	if foundVariant != "" {
-		// we don't do a full reconciliation (update push app on UPS server based on CR content) but we
-		// only do initial creation of push apps.
-		reqLogger.Info("Skip reconcile: Android Variant already exists in UPS", "AndroidVaraint.Name", instance.Name)
-		return reconcile.Result{}, nil
+	if (foundVariant == unifiedpush.AndroidVariant{}) {
+		createdVariant, err := unifiedpushClient.CreateAndroidVariant(instance)
+		if err != nil {
+			reqLogger.Error(err, "Error creating Android variant in UPS.", "AndroidVaraint.Name", instance.Name)
+			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		instance.ObjectMeta.Annotations["VariantId"] = createdVariant.VariantId
```

Comment:
```suggestion
		instance.ObjectMeta.Annotations["variantId"] = createdVariant.VariantId
```

I'll make this change & same for other annotations modified here.